src/pages/Appointments/AppointmentsPage.tsx

```diff
@@ -360,100 +358,23 @@ export default function AppointmentsPage(props: { facilityId?: string }) {
-                        : t("no_results")}
-                    </CommandEmpty>
-                    <CommandGroup>
-                      <PopoverClose className="w-full">
-                        <CommandItem
-                          value="all"
-                          onSelect={() =>
-                            setQParams({
-                              ...qParams,
-                              practitioner: null,
-                              slot: null,
-                            })
-                          }
-                          className="cursor-pointer"
-                        >
-                          <span>{t("show_all")}</span>
-                          {!qParams.practitioner && (
-                            <CheckIcon className="ml-auto" />
-                          )}
-                        </CommandItem>
-                      </PopoverClose>
-                      {schedulableUsersQuery.data?.users.map((user) => (
-                        <PopoverClose className="w-full" key={user.id}>
-                          <CommandItem
-                            value={formatName(user)}
-                            onSelect={() =>
-                              setQParams({
-                                ...qParams,
-                                practitioner: user.username,
-                                slot: null,
-                              })
-                            }
-                            className="cursor-pointer"
-                          >
-                            <div className="flex items-center gap-2">
-                              <Avatar
-                                imageUrl={user.profile_picture_url}
-                                name={formatName(user)}
-                                className="size-6 rounded-full"
-                              />
-                              <span>{formatName(user)}</span>
-                              <span className="text-xs text-gray-500 font-medium">
-                                {user.user_type}
-                              </span>
-                            </div>
-                            {qParams.practitioner === user.username && (
-                              <CheckIcon className="ml-auto" />
-                            )}
-                          </CommandItem>
-                        </PopoverClose>
-                      ))}
-                    </CommandGroup>
-                  </CommandList>
-                </Command>
-              </PopoverContent>
-            </Popover>
+            <UserAutocomplete
+              value={practitioner || undefined}
+              id="practitioner"
+              options={{ users: resources ?? [] }}
+              onSelect={(user) => {
+                if (user) {
+                  setQParams({
+                    ...qParams,
+                    practitioner: user.username,
+                    slot: null,
+                  });
+                }
```

Comment:
so what happens when i click "show all" or unselect the user? logic for clearing the practicioner filter from query params is not written...